packages/@react-spectrum/s2/src/SelectBoxGroup.tsx

```diff
@@ -0,0 +1,256 @@
+  styles?: StyleProps,
+  onChange: (value: SelectBoxValueType) => void,
+  value?: SelectBoxValueType,
+  defaultValue?: SelectBoxValueType,
+  isRequired?: boolean,
+  isDisabled?: boolean,
+  label?: ReactNode
+}
+
+/**
+ * SelectBox groups allow users to select one or more options from a list.
+ * All possible options are exposed up front for users to compare.
+ */
+
+export const SelectBoxGroup = /*#__PURE__*/ forwardRef(function SelectBoxGroup(props: SelectBoxGroupProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, SelectBoxGroupContext);
+  
+  let {
+    label,
+    children,
+    onSelectionChange,
+    defaultValue,
+    selectionMode = 'single',
+    size = 'M',
+    orientation = 'vertical',
+    isEmphasized,
+    numColumns = 2,
+    gutterWidth = 'default',
+    isRequired = false,
+    isDisabled = false,
+    UNSAFE_style
+  } = props;
+
+  const allowMultiSelect = selectionMode === 'multiple';
+  
+  const domRef = useDOMRef(ref);
+
+  const getChildrenToRender = () => {
+    const childrenToRender = React.Children.toArray(children).filter((x) => x);
+    try {
+      const childrenLength = childrenToRender.length;
+      if (childrenLength <= 0) {
+        throw new Error('Invalid content. SelectBox must have at least a title.');
+      }
+      if (childrenLength > 9) {
+        throw new Error('Invalid content. SelectBox cannot have more than 9 children.');
+      }
+    } catch (e) {
+      console.error(e);
+    }
+    return childrenToRender;
+  };
+
+  // Context value
+  const selectBoxContextValue = useMemo(
+    () => ({
+      allowMultiSelect,
+      size,
+      orientation,
+      isEmphasized
+    }),
+    [allowMultiSelect, size, orientation, isEmphasized]
+  );
+
+  return (
+    <SelectorGroup
+      allowMultiSelect={allowMultiSelect}
+      value={props.value}
+      defaultValue={defaultValue}
+      onChange={onSelectionChange}
+      isRequired={isRequired}
+      isDisabled={isDisabled}
+      label={label}
+      ref={domRef}
+      UNSAFE_style={UNSAFE_style}>
+      <div
+        className={gridStyles({gutterWidth, orientation}, props.styles)}
+        style={{
+          gridTemplateColumns: `repeat(${numColumns}, 1fr)`
+        }}>
+        <SelectBoxContext.Provider value={selectBoxContextValue}>
+          {getChildrenToRender().map((child) => {
+            return child as ReactElement;
+          })}
+        </SelectBoxContext.Provider>
+      </div>
+    </SelectorGroup>
+  );
+});
+
+const SelectorGroup = forwardRef<HTMLDivElement, SelectorGroupProps>(function SelectorGroupComponent({
+  allowMultiSelect,
+  children,
+  UNSAFE_className,
+  onChange,
+  value,
+  defaultValue,
+  UNSAFE_style,
+  isRequired,
+  isDisabled,
+  label
+}, ref) {
+  const baseProps = { 
+    isRequired,
+    isDisabled,
+    UNSAFE_className,
+    UNSAFE_style,
+    children,
+    onChange,
+    ref
+  };
+
+  return allowMultiSelect ? (
```

Comment:
part of me wonders if this should be built using Checkbox/Radios. i say that because the keyboard experience is kinda interesting? for example, it's layout is sorta grid-like, so i would expect that if i arrow down, i would go to the item below but sometimes i go the item to the right due to the nature of how Checkbox/Radio's work